src/java/org/apache/nutch/fetcher/FetcherThread.java

```diff
@@ -695,14 +696,23 @@ private ParseStatus output(Text key, CrawlDatum datum, Content content,
           int validCount = 0;
 
           // Process all outlinks, normalize, filter and deduplicate
+          
+          // NUTCH-2413 Apply filters or normalizers only if configured
+          URLFilters urlFiltersForOutlinks = null;
+          if (conf.getBoolean("parse.filter.urls", true))
+            urlFiltersForOutlinks = urlFilters;
+          URLNormalizers normalizersForOutlinks = null;
+          if (conf.getBoolean("parse.normalize.urls", true))
+            normalizersForOutlinks = normalizers;
+          
           List<Outlink> outlinkList = new ArrayList<>(outlinksToStore);
           HashSet<String> outlinks = new HashSet<>(outlinksToStore);
           for (int i = 0; i < links.length && validCount < outlinksToStore; i++) {
             String toUrl = links[i].getToUrl();
-
+                      
             toUrl = ParseOutputFormat.filterNormalize(url.toString(), toUrl,
                 origin, ignoreInternalLinks, ignoreExternalLinks, ignoreExternalLinksMode,
-                    urlFilters, urlExemptionFilters,  normalizers);
+                urlFiltersForOutlinks, urlExemptionFilters,  normalizersForOutlinks);
```

Comment:
Please, keep the indentation or use the [Eclipse code formatting rules](https://github.com/apache/nutch/blob/master/eclipse-codeformat.xml).